framework/src/constraints/MortarConstraintBase.C

```diff
@@ -115,7 +120,9 @@ MortarConstraintBase::MortarConstraintBase(const InputParameters & parameters)
     _tangents(_assembly.tangents()),
     _coord(_assembly.mortarCoordTransformation()),
     _q_point(_assembly.qPointsMortar()),
-    _test(_var ? _var->phiLower() : _test_dummy),
+    _use_petrov_galerkin(getParam<bool>("use_petrov_galerkin")),
+    _aux_lm_var(isCoupled("aux_lm") ? getVar("aux_lm", 0) : nullptr),
+    _test(_var ? (_use_petrov_galerkin ? _aux_lm_var->phiLower() : _var->phiLower()) : _test_dummy),
```

Comment:
Would we get a seg fault here if the user sets up the boolean but forgets to set the coupled auxiliary variable?

Reply:
Yeah that's a good point. I am not entirely sure what's the best practice here?

Reply:
I'd suggest to check within the ternary operators  that `_aux_lm_var` is not a `nullptr` in addition to `_use_petrov_galerkin`, so if the user chose to have PG but didn't provide an auxiliary variable, `_test` would just be `_var->phiLower()`. The rest of the user input consistency requirements should be taken care of by the checks you already have in the constructor.

Reply:
> so if the user chose to have PG but didn't provide an auxiliary variable, _test would just be _var->phiLower()

We should throw an error here, too, I suppose?

Reply:
Yes, we should throw a `paramError` but avoid a segmentation fault in the initialization list.

Reply:
Agreed. thanks for the clarification!